infor-design/enterprise
# 1865 - Fix a few issues with notifications involving styles and structure

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
We needed to replace the rgba notification background colors with hexes. This will require `ids-identity@2.5.0` to be released.

Since I was under the hood, I also updated the structure to fix #1573 and made the responsive view more "flexible".
- `html` - changed the`<span>` to a `<p>` for wrapping
- `css` - removed `line-height` style in favor of padding around the whole thing to allow text wrapping
- `css` - updated tokens with specific notification tokens
- `css` - removed conflicting positioning of icons
- `js` - separated the js code into smaller string snippets
- `js` - removed the string with interpolation of an if-statement with inside strings with interpolation
- `js` - included the`<a>` inside of the `<p>` tag for semantics and proper wrapping
- `demo`- added a long notification to test wrapping

**Related github/jira issue (required)**:
- Fixes https://github.com/infor-design/enterprise/issues/1857
- Fixes https://github.com/infor-design/enterprise/issues/1865
- Fixes https://github.com/infor-design/enterprise/issues/1573

**Steps necessary to review your pull request (required)**:
1. 📝  `npm i && npm run start`
1. 📝 Testing #1857 and #1865:
    - Go to [localhost notifications](http://localhost:4000/components/notification/example-index.html?layout=embedded)
    - verify that the colors match up to the description in [this comment](https://github.com/infor-design/design-system/issues/294#issuecomment-476274897) and in [this pull request](https://github.com/infor-design/design-system/pull/371)
    - verify the colors are "bolder" than [4.17.1](https://4171-enterprise.demo.design.infor.com/components/notification/example-index.html)
    - verify the responsive views wrap text an visually look up to par (no overlapping of text/icons, etc)
1. 📝 Testing #1573
    - In Firefox, go to [localhost notifications](http://localhost:4000/components/notification/example-index.html) 
    - verify you can click the close `x` button
    - bonus is that resizing the browser will adjust for long notifications w/o an ellipsis ([compare to current responsive here]([4.17.1](https://4171-enterprise.demo.design.infor.com/components/notification/example-index.html))

ToDo:
☑️  A note to the change log
☑️  Need `ids-identity@2.5.0` to be released
